src/main/host/syscall/socket.c

```diff
@@ -725,6 +725,12 @@ SysCallReturn syscallhandler_bind(SysCallHandler* sys,
         return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = -EINVAL};
     }
 
+    /* Make sure the addr PluginPtr is not NULL. */
+    if (!addrPtr.val) {
```

Comment:
I think we don't need this check, since the MemoryManager will detect if the pointer isn't accessible?

Reply:
Removed this check since it isn't required with the change from your following comment.